src/ptf/testutils.py

```diff
@@ -2245,15 +2245,18 @@ def dp_poll(test, device_number=0, port_number=None, timeout=-1, exp_pkt=None):
         test.at_receive(result.packet, device_number=result.device, port_number=result.port)
     return result
 
-def verify_packet(test, pkt, port_id):
+def verify_packet(test, pkt, port_id, timeout=2):
     """
     Check that an expected packet is received
     port_id can either be a single integer (port_number on default device 0)
     or a tuple of 2 integers (device_number, port_number)
     """
+    if timeout==None:
```

Comment:
actually one more thing
does this if statement really makes sense? I don't remember seeing a similar one anywhere else. For it to be `None` the user would have to pass `None` explicitly. It doesn't correspond to any kind of "default" case, since timeout already defaults to 2.

Reply:
If the user passes nothing to verify_packets() function by default it will be "None". verify_packets() will call verify_packet(, timeout=None). The default will not kick in because the caller explicitly asked for None. So I had to add the if there.